pkg/listeners/listeners_test.go

```diff
@@ -0,0 +1,142 @@
+// +build linux,cgo
+
+package listeners
+
+import (
+	"crypto/tls"
+	"flag"
+	"net"
+	"os"
+	"strconv"
+	"testing"
+
+	apiserver "github.com/docker/docker/api/server"
+	"github.com/docker/docker/cmd/dockerd/hack"
+	"github.com/docker/docker/dockerversion"
+	"github.com/docker/go-connections/tlsconfig"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var rootEnabled = false
+
+func init() {
+	flag.BoolVar(&rootEnabled, "test.root", false, "enable tests that require root")
+}
+func RequiresRoot(t *testing.T) {
+	if !rootEnabled {
+		t.Skip("skipping test that requires root")
+		return
+	}
+	require.Equal(t, 0, os.Getuid(), "This test must be run as root.")
+}
+func TestInitForUnix(t *testing.T) {
+	RequiresRoot(t)
```

Comment:
The Init function present in `listeners_unix.go` calls `sockets.NewUnixSocket` function, this function internally calls `os.Chown` which requires root permissions to execute. Otherwise it returns a `Permission Denied` error.